blockchain/chain_test.go

```diff
@@ -1311,3 +1313,381 @@ func TestIsAncestor(t *testing.T) {
 			branch2Nodes[0].hash.String())
 	}
 }
+
+func TestInvalidateBlock(t *testing.T) {
+	tests := []struct {
+		name     string
+		chainGen func() (*BlockChain, []*chainhash.Hash, func())
+	}{
+		{
+			name: "One branch, invalidate once",
+			chainGen: func() (*BlockChain, []*chainhash.Hash, func()) {
```

Comment:
think this should be functions below rather than explicitly in full form here

Reply:
Could you clarify what you mean here?

Reply:
I just mean moving the `chainGen` function into a function outside of the test so you just call `chainGen:funcHere` for readability. Just my preference, but it's non blocking